sdks/python/apache_beam/testing/test_utils.py

```diff
@@ -129,3 +131,47 @@ def delete_files(file_paths):
     raise RuntimeError('Clean up failed. Invalid file path: %s.' %
                        file_paths)
   FileSystems.delete(file_paths)
+
+
+def wait_for_subscriptions_created(subs, timeout=60):
+  """Wait for all PubSub subscriptions are created."""
+  return _wait_until_all_exist(subs, timeout)
+
+
+def wait_for_topics_created(topics, timeout=60):
+  """Wait for all PubSub topics are created."""
+  return _wait_until_all_exist(topics, timeout)
+
+
+def _wait_until_all_exist(components, timeout):
+  need_wait = set(components)
+  start_time = time.time()
+  while time.time() - start_time <= timeout:
+    for c in components:
+      if c in need_wait and c.exists():
+        need_wait.remove(c)
+    if len(need_wait) == 0:
+      return True
+    time.sleep(2)
+
+  raise RuntimeError('Timeout after %d seconds. %d of %d topics/subscriptions '
```

Comment:
Would it be helpful if you log what does not exist?

Reply:
sure. done.